### ISSUE TITLE
508-defect-3: Downloading letters should be styled and written as links (but may be a button when generating)

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#45998]
## Point of contact

Josh Kim

## List of issues by severity
There are 5 tickets related to this issue: 
Main ticket: https://github.com/department-of-veterans-affairs/va.gov-team/issues/45650
Heading: https://github.com/department-of-veterans-affairs/va.gov-team/issues/45944
Labelling: https://github.com/department-of-veterans-affairs/va.gov-team/issues/45945
Responsive: https://github.com/department-of-veterans-affairs/va.gov-team/issues/45946
Download: 
I've grouped these issues together as they're closely related and dependent on one another. Whatever solution we consider will likely need to take into account all of these issues together from a holistic perspective @skylerschain 

### Download button needs review (Moderate)
If the user is downloading pdfs that have already been generated, they should be styled and [written as links following design system guidance](https://design.va.gov/components/link/#download).

If the user is _generating_ a pdf, and then downloading it, we should investigate if we can pre-generate the pdfs (and their variations) so that they may be styled as links. Otherwise we may have to style and code them as buttons.

We should try our best to use links (instead of buttons) are preferred as they can provide the user additional context and actions including:
- opening in the browser or in a new tab
- a count of pages
- the type of document it is (PDF, word doc, etc)
- how large it is (for users with low bandwidth connections or limited data)

https://user-images.githubusercontent.com/14154792/184921694-3315ebc4-e781-447b-a942-357839887299.MP4

## Reproduction steps

1. [Log in as user 299 and get to step 2](https://staging.va.gov/records/download-va-letters/letters/letter-list)
2. Open the benefit summary accordion

## Proposed solution or next steps

TBD, this requires further exploration with design, engineering, and content. My immediate thoughts are:
- we should look into past research on why the designers choose the accordion and table pattern (for example, how complex can these tables be? e.g. for military service history); if no strong design reasons exist, we may want to consider redesigning this
- we should consider using plain checkboxes (instead of checkboxes within a table)
- depending on the length of this, it may deserve to live on a separate page 

## Type of issue

- [x] Screenreader
- [x] Keyboard
- [ ] Focus
- [x] Headings
- [ ] Color
- [x] Zoom
- [x] Semantics
- [ ] Axe-core
- [ ] Something else

**Comment by jacobworrell** [comment]
@joshkimux let's discuss this one. We could change the button to a link even though it's technically generating a file, but we're not sure if that's actually "better" all things holistically. If we don't have a URL prior to clicking the link, we'd probably have to default to the button.

**Comment by joshkimux** [comment]
@jacobworrell sounds like a plan!

**Comment by sortizsh** [comment]
We may need to make an exception for this defect due to balancing other UX concerns.

**Comment by sortizsh** [comment]
@jaredcunha  FYI

**Comment by jaredcunha** [comment]
Spoke with @joshkimux and we have 3 options as far as what to do:

**Option 1: Generate pdfs on page load**
If we generate the PDFs on page load, we can provide an `href` to point a link to. Doing this would allow us to swap the button for an action link, and otherwise not make any other changes to the UI or the UX. However, there are questions we would need to answer before pursuing this option:
1. How might this impact performance?
2. Are there potential issues with consent?
3. What might this do to the file size?
4. How might this impact users with low bandwidth?

There are probably other questions we may want to ask as well.

**Option 2: One button to generate the pdf, then inject a link to download it.**
This approach avoids the need to pre-generate a file and use semantic HTML elements as they are intended. However, I don't anyone would disagree that this changes the UX significantly, and negatively. We dig further to understand the cost of this of approach, but much of that might be self-evident.

**Option 3: Do nothing**
Don't touch, and leave as is. This would be the option we might need to choose after some discovery on the first two options.

@joshkimux lemme know if I captured anything incorrectly.

**Comment by joshkimux** [comment]
This is perfect @jaredcunha ! ❤️

**Comment by skylerschain** [comment]
Yeah I'm for Option 1 if it doesn't negatively impact performance. Styling these as download links similar to what we're doing for decision letters seems like the best UI choice.

**Comment by sortizsh** [comment]
band aid: address width issue of button @sortizsh

**Comment by jacobworrell** [comment]
We acknowledge this as an issue, a legit accessibility defect, but any feasible "solution" we've come up with is worse than the problem by introducing other defects and Ux issues.

**Comment by joshkimux** [comment]
@jacobworrell thank you for the update, appreciate you and the team's hard work on this. Would it be alright if I...
1. removed your team's labels
2. added a DST label
3. re-opened this 
4. mark is as blocked

Context: want to keep this discoverable into the future as a known issue that may emerge on other teams (and should possibly get written guidance in the design system). 

Alternatively, I can open up a separate ticket it it breaks anything in terms of tracking!
